lib/modules/manager/poetry/extract.ts

```diff
@@ -14,13 +14,27 @@ import type { PackageDependency, PackageFile } from '../types';
 import { extractLockFileEntries } from './locked-version';
 import type { PoetryFile, PoetrySection } from './types';
 
+class PoetryGroupSection {
+  constructor(public name: string) {}
+}
+
 function extractFromSection(
   parsedFile: PoetryFile,
-  section: keyof Omit<PoetrySection, 'source'>,
+  section: keyof Omit<PoetrySection, 'source' | 'group'> | PoetryGroupSection,
```

Comment:
i think we should better split this function, so we don't need the extra class and conditions

Reply:
https://github.com/renovatebot/renovate/pull/17964/commits/a0d64ba4bd9824e9c0dc15de993853f6b522d62c. Not sure if this is exactly what you had in mind, so tell me if not.